### PR TITLE
feat: update Keycloak plugin for dynamic backend

### PR DESCRIPTION
**Pull request by craigtracey**
Add dynamic backend module capabilities for the Keycloak backend plugin.

### CODE REVIEW

File: plugins/keycloak-backend/CHANGELOG.md
Comment (tumido) [inline plugins/keycloak-backend/CHANGELOG.md:1]: Can you please undo the CHANGELOG changes? 🙂

File: plugins/keycloak-backend/package.json
Comment (tumido) [inline plugins/keycloak-backend/package.json:1]: Can you please also add the `export-dynamic` script, so we can actually export that dynamic package?

File: plugins/keycloak-backend/package.json [hunk 1]
```diff
@@ -11,7 +11,22 @@
     "types": "dist/index.d.ts"
   },
   "backstage": {
-    "role": "backend-plugin"
+    "role": "backend-plugin-module"
```
Comment (tumido) [inline plugins/keycloak-backend/package.json:14]: You are right about this! Correct! 🙂

**Review by tumido — CHANGES_REQUESTED** [review]

**Comment by tumido** [comment]
Thank you for your PR, @craigtracey ! 👍 I see only minor nits, nothing major 👍 🙂

File: plugins/keycloak-backend/CHANGELOG.md
Comment (craigtracey) [reply plugins/keycloak-backend/CHANGELOG.md:1]: These were coming from failing Prettier tests. Will undo and see if it passes.

File: plugins/keycloak-backend/package.json
Comment (craigtracey) [reply plugins/keycloak-backend/package.json:1]: As this is new, I am not familiar with it yet. Will take a look and attempt to incorporate it.

**Review by davidfestal — CHANGES_REQUESTED** [review]

File: plugins/ocm/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@janus-idp/backstage-plugin-ocm",
-  "version": "3.2.3",
+  "version": "3.3.0",
```
Comment (davidfestal) [inline plugins/ocm/package.json:3]: Why this change ?

**Review by davidfestal — CHANGES_REQUESTED** [review]
This PR has adds support for the Backstage new backend system.
However this is not sufficient to add support for the Dynamic backend plugins loading in the Janus Showcase applcation, since the Janus `backstage-showcase` is still based on the legacy backend system, until the new backend leaves the alpha state.
So for now you have to add some additional entrypoints that will provide dynamic loading logic for both the old and the new backend system.
The code to add in the Keycloak case can be found in the [dynamic folder of the corresponding showcase wrapper](https://github.com/janus-idp/backstage-showcase/tree/main/dynamic-plugins/wrappers/janus-idp-backstage-plugin-keycloak-backend-dynamic/src/dynamic)

File: plugins/ocm/package.json (hunk 2 above)
Comment (craigtracey) [reply plugins/ocm/package.json]: This was an artifact of a bad rebase.

File: plugins/keycloak-backend/package.json [hunk 3]
```diff
@@ -38,11 +55,13 @@
   },
   "devDependencies": {
     "@backstage/backend-app-api": "0.5.6",
+    "@backstage/backend-test-utils": "^0.2.7",
     "@backstage/cli": "0.23.0",
     "@backstage/plugin-auth-node": "0.4.0",
     "@backstage/plugin-catalog-backend": "1.14.0",
     "@backstage/plugin-permission-common": "0.7.9",
     "@backstage/plugin-permission-node": "0.7.17",
+    "@dfatwork-pkgs/backstage-cli": "0.22.9-next.6",
```
[the remark this answers is not in the document]
Comment (craigtracey) [reply plugins/keycloak-backend/package.json]: Done

File: plugins/keycloak-backend/package.json [hunk 4]
```diff
@@ -23,10 +38,12 @@
     "test": "backstage-cli package test --passWithNoTests --coverage",
     "clean": "backstage-cli package clean",
     "prepack": "backstage-cli package prepack",
-    "postpack": "backstage-cli package postpack"
+    "postpack": "backstage-cli package postpack",
+    "export-dynamic": "backstage-cli package export-dynamic-plugin --embed-package @backstage/plugin-explore-backend --embed-package @backstage/plugin-search-backend-module-explore"
```
[the remark this answers is not in the document]
Comment (craigtracey) [reply plugins/keycloak-backend/package.json]: Done

File: plugins/keycloak-backend/package.json
Comment (craigtracey) [reply plugins/keycloak-backend/package.json:1]: Added

File: plugins/keycloak-backend/CHANGELOG.md
Comment (craigtracey) [reply plugins/keycloak-backend/CHANGELOG.md:1]: Reverted

**Comment by craigtracey** [comment]
> This PR has adds support for the Backstage new backend system. However this is not sufficient to add support for the Dynamic backend plugins loading in the Janus Showcase applcation, since the Janus `backstage-showcase` is still based on the legacy backend system, until the new backend leaves the alpha state. So for now you have to add some additional entrypoints that will provide dynamic loading logic for both the old and the new backend system. The code to add in the Keycloak case can be found in the [dynamic folder of the corresponding showcase wrapper](https://github.com/janus-idp/backstage-showcase/tree/main/dynamic-plugins/wrappers/janus-idp-backstage-plugin-keycloak-backend-dynamic/src/dynamic)

I am not familiar with how Janus Showcase fits in here. As I look at Janus Showcase, it looks as if the keycloak backend has already been moved to the new backend system. Does that work make this PR obsolete?

**Comment by davidfestal** [comment]
> > This PR has adds support for the Backstage new backend system. However this is not sufficient to add support for the Dynamic backend plugins loading in the Janus Showcase applcation, since the Janus `backstage-showcase` is still based on the legacy backend system, until the new backend leaves the alpha state. So for now you have to add some additional entrypoints that will provide dynamic loading logic for both the old and the new backend system. The code to add in the Keycloak case can be found in the [dynamic folder of the corresponding showcase wrapper](https://github.com/janus-idp/backstage-showcase/tree/main/dynamic-plugins/wrappers/janus-idp-backstage-plugin-keycloak-backend-dynamic/src/dynamic)
> 
> I am not familiar with how Janus Showcase fits in here. As I look at Janus Showcase, it looks as if the keycloak backend has already been moved to the new backend system. Does that work make this PR obsolete?

It doesn't make this PR at all.
Dynamic plugins in the Janus Showcase has been designed to support both the old backend system and the new backend system (at least until the new backend system is completely out of alpha status).
This is why There are some additions required on top of what you already provided.
But this would be part of a followup PR that would complement this one.

Let's merge your PR already.

**Review by davidfestal — APPROVED** [review]

**Pull request merged**